106-list-unregister-nodes-admin-api.md

```diff
@@ -0,0 +1,92 @@
+It replaces the current mechanism that tracks registered nodes in the `Kafka` custom resource status.
+
+## Current situation
+
+Currently, the Strimzi operator tracks the list of registered KRaft nodes (both brokers and controllers) in the `Kafka` custom resource status via the `registeredNodeIds` field, as shown in the example below:
+
+```yaml
+status:
+    clusterId: bXz0umhlQ0WLSobuC-Irdg
+    #....
+    registeredNodeIds:
+    - 0
+    - 1
+    - 2
+    - 3
+    - 4
+    - 5
+```
+
+When the Kafka cluster is scaled down, the operator compares the current running node IDs against the stored `registeredNodeIds` to determine which nodes should be unregistered via the Kafka Admin API, because they were removed.
+This logic was introduced as part of the [Strimzi proposal 081](https://github.com/strimzi/proposals/blob/main/081-unregistration-of-KRaft-nodes.md), which implemented a workaround for the limited node unregistration support in Kafka at that time. There was no reliable way to list registered nodes (especially fenced ones), so Strimzi maintained this state internally.
+
+However, this approach has limitations.
+For example, when brokers are initially deployed as mixed-mode nodes (controller + broker) and later have their broker role removed, they remain registered.
+Although these nodes become fenced and controller-only, they may still receive partition assignments for new topics.
+The current implementation does not detect or handle this scenario, leading to missing unregistration.
+This issue has been reported by the community in [GitHub issue #11477](https://github.com/strimzi/strimzi-kafka-operator/issues/11477).
+
+## Motivation
+
+With the introduction of [KIP-1073](https://cwiki.apache.org/confluence/display/KAFKA/KIP-1073%3A+Return+fenced+brokers+in+DescribeCluster+response) in Apache Kafka 4.0.0, the Kafka Admin API now provides the ability to list all registered brokers, including fenced ones.
+This makes it possible to manage unregistration entirely through standard Kafka APIs.
+
+This removes the need for Strimzi to track nodes registration status manually and enables proper handling of edge cases like mixed-node role changes.
+As a result, it also allows for a proper resolution of issue [#11477](https://github.com/strimzi/strimzi-kafka-operator/issues/11477).
+
+## Proposal
+
+This proposal suggests removing the custom registration tracking and instead relying on Apache Kafka's built-in registration tracking available via the Admin API.
+
+### Key changes
+
+The key changes are:
+
+* remove the `status.registeredNodeIds` field from the `Kafka` custom resource. The operator will no longer maintain its own list of registered nodes.
```

Comment:
You probably should not remove it for backwards compatibility reasons. Just stop using it and deprecate it. Maybe you did not meant it that way, but it probably deserves more clarity.

Reply:
To be honest I meant that but yes I am wrong. We need to deprecate it first. We also stated it in the proposal 081. Thanks for pointing this out.